File: canFormPalindrome/canFormPalindrome.cpp
#include <iostream>
#include <string>
using namespace std;

bool canFormPalindrome(const string& s) {
    int count[256] = {0}; // Tạo mảng đếm với kích thước bảng mã ASCII

    // Đếm tần suất của mỗi ký tự
    for (char c : s) {
        count[c]++;
    }

    int oddCount = 0;

    // Kiểm tra số lượng ký tự có tần suất lẻ
    for (int i = 0; i < 256; i++) {
        if (count[i] % 2 != 0) {
            oddCount++;
        }

        // Nếu có nhiều hơn một ký tự có tần suất lẻ, chuỗi không thể là đối xứng
        if (oddCount > 1) {
            return false;
        }
    }
    return true;
}

int main() {
    string s;
    cout << "Enter the string: ";
    cin >> s;
    if (canFormPalindrome(s)) {
        cout << "Yes";
    } else {
        cout << "No";
    }
    return 0;
}

File: canFormPalindrome/canFormPalindrome.py
def can_form_palindrome(s):
    count = [0] * 256  # Tạo mảng đếm với kích thước bảng mã ASCII

    # Đếm tần suất của mỗi ký tự
    for char in s:
        count[ord(char)] += 1

    odd_count = 0

    # Kiểm tra số lượng ký tự có tần suất lẻ
    for c in count:
        if c % 2 != 0:
            odd_count += 1

        # Nếu có nhiều hơn một ký tự có tần suất lẻ, chuỗi không thể là đối xứng
        if odd_count > 1:
            return False
    return True

# Nhập chuỗi từ người dùng và kiểm trageeksforgeeks
s = input("Enter the string: ")
if can_form_palindrome(s):
    print("Yes")
else:
    print("No")
